src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -293,9 +294,93 @@ class StatementSegment(BaseSegment):
             Ref("TableAttributeSegment"),
             Ref("ColumnAttributeSegment"),
             Ref("ColumnEncodingSegment"),
+            Ref("CreateUserSegment"),
+            Ref("CreateGroupSegment"),
         ],
     )
 
     match_grammar = redshift_dialect.get_segment(
         "StatementSegment"
     ).match_grammar.copy()
+
+
+@redshift_dialect.segment()
+class CreateUserSegment(BaseSegment):
+    """`CREATE USER` statement.
+
+    https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_USER.html
+    """
+
+    type = "create_user"
+
+    match_grammar = StartsWith(
+        Sequence("CREATE", "USER"),
+    )
+    parse_grammar = Sequence(
+        "CREATE",
+        "USER",
```

Comment:
@WittierDinosaur and I [had a big discussion on this](https://github.com/sqlfluff/sqlfluff/pull/1877#discussion_r746851959) and came to the conclusion that `match_grammar`'s that start with the same as `parse_grammar`'s doesn't add much and so there's no real benefit for this (at least I think that's where we landed!).

So I'd suggest simplifying this to:

```suggestion
    match_grammar = Sequence(
        "CREATE",
        "USER",
```

Reply:
@tunetheweb makes sense, I only did it because it's a very common pattern in the existing dialect. Will remove from here.

Should I make a separate PR to start removing this from the rest of the grammar as well to tidy things up?

Reply:
I actually checked a few statements in the Redshift dialogue and saw it wasn’t used on the ones I checked. Was going to let you away with it if it was the standard in this dialect. Guess you and I happened to check different statements! But then wouldn’t StartsWith already exist in the dialect? 🤔

@WittierDinosaur what do you reckon? Clean it up, or you still convinced this is the way to go and eventually you might make it work (in which case I’d suggest not cleaning it up but also not encouraging it’s use in case we never do that).

Reply:
I would've looked at a similar segment from another dialect, but the question still stands for those other dialects 😄